conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureClientEndpoint.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.lib.internal;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Signifies that the given method corresponds to a conjure endpoint with the given http path and method.
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.METHOD)
+public @interface ConjureClientEndpoint {
+    String method();
+
+    String path();
```

Comment:
Dunno if the coupling is right, but there's HttpMethod in com.palantir.conjure.spec

Reply:
I want to keep type simple for the annotation. As it is the only place where this gets filled in uses the definition.getHttpMethod().get() to populate, so I expect that is fine.

Reply:
Agree that method is simple enough here, and we don't want conjure generated code to have a dependency onto the conjure spec types.